### ISSUE TITLE
VidCoder 6.22 Beta: Subtitles names are not assigned to the encoded movie

### ISSUE THREAD

**Issue by dan64** [RandomEngy/VidCoder#753]
One of the new feature of VidCoder 6.22 Beta is to assign (or change) the subtitles names. I tried to use this feature by encoding a h.264 mkv movie in h.265 mkv movie. But the final encoded movies is without the assigned names.

For example I assigned the following names:

![Subtitle](https://user-images.githubusercontent.com/5787502/107879277-248ace80-6ed8-11eb-930f-b2f8017b9627.PNG)

but the mediainfo output don't shows these names:

`Text #1
ID                          : 4
Format                      : ASS
Codec ID                    : S_TEXT/ASS
Codec ID/Info               : Advanced Sub Station Alpha
Duration                    : 2 h 27 min
Compression mode            : Lossless
Language                    : Italian
Default                     : Yes
Forced                      : No

Text #2
ID                          : 5
Format                      : ASS
Codec ID                    : S_TEXT/ASS
Codec ID/Info               : Advanced Sub Station Alpha
Duration                    : 2 h 27 min
Compression mode            : Lossless
Language                    : Italian
Default                     : No
Forced                      : No

Text #3
ID                          : 6
Format                      : ASS
Codec ID                    : S_TEXT/ASS
Codec ID/Info               : Advanced Sub Station Alpha
Duration                    : 2 h 27 min
Compression mode            : Lossless
Language                    : English
Default                     : No
Forced                      : No

Text #4
ID                          : 7
Format                      : ASS
Codec ID                    : S_TEXT/ASS
Codec ID/Info               : Advanced Sub Station Alpha
Duration                    : 2 h 27 min
Compression mode            : Lossless
Language                    : English
Default                     : No
Forced                      : No
`

**Comment by quebecman** [comment]
Strange this works for me.

**Comment by dan64** [comment]
I attached the extended log of the encoding. It is possible to see that at rows 750-800 are specified the encoding settings. For the audio tracks is specified the name, and in effect that audio tracks are correctly renamed, but for subtitles it is missing any assignment and in output they are without name.  

[2021-02-14 Encode WW84.2020.1080p.x264 (Encoded).mkv-log.zip](https://github.com/RandomEngy/VidCoder/files/5977978/2021-02-14.Encode.WW84.2020.1080p.x264.Encoded.mkv-log.zip)

**Comment by quebecman** [comment]
I have looked at the logs.  Is there something else I need to do to help?

________________________________

**Comment by dan64** [comment]
I encoded another clip with the same issue. The VidCoder settings are the following

![Subtitle](https://user-images.githubusercontent.com/5787502/107881433-9ff27d00-6ee4-11eb-9205-16aee788d453.PNG)

The forced subtitle is defined as default track on the source container, while VidCoder fail to set this assign this attribute to the FORCED subtitle. The encoded clip has the same problems: audio naming is working, subtitle naming not.

I attached the encoding log. The clip used can be downloaded from the following link:

[https://filebin.net/w6oi2calazeqg4nh](url)

[2021-02-14 Encode WW84.2020.1080p.x265 (Encoded).mkv-log.zip](https://github.com/RandomEngy/VidCoder/files/5978007/2021-02-14.Encode.WW84.2020.1080p.x265.Encoded.mkv-log.zip)

**Comment by quebecman** [comment]
Thanx for your help DAN.

________________________________

**Comment by RandomEngy** [comment]
What exact steps are you using to start the encode? Are you pressing "Add to Queue", then "Encode"? If you just drag a file to the queue area it won't pick up the naming you entered.

Yours looks like this:

```
    "SubtitleList": [
      {
        "Burn": false,
        "Default": false,
        "Forced": false,
        "ID": 1,
        "Offset": 0,
        "Track": 0
      },
      {
        "Burn": false,
        "Default": false,
        "Forced": false,
        "ID": 2,
        "Offset": 0,
        "Track": 1
      },
      {
        "Burn": false,
        "Default": false,
        "Forced": false,
        "ID": 3,
        "Offset": 0,
        "Track": 2
      },
      {
        "Burn": false,
        "Default": false,
        "Forced": false,
        "ID": 4,
        "Offset": 0,
        "Track": 3
      }
    ]
```

No names present.

When I do the same, it looks like this:

```
    "SubtitleList": [
      {
        "Burn": false,
        "Default": false,
        "Forced": false,
        "ID": 1,
        "Offset": 0,
        "Track": 0,
        "Name": "EngCust"
      },
      {
        "Burn": false,
        "Default": false,
        "Forced": false,
        "ID": 2,
        "Offset": 0,
        "Track": 1,
        "Name": "EspCust"
      },
      {
        "Burn": false,
        "Default": false,
        "Forced": false,
        "ID": 3,
        "Offset": 0,
        "Track": 2,
        "Name": "EngCust2"
      },
      {
        "Burn": false,
        "Default": false,
        "Forced": false,
        "ID": 4,
        "Offset": 0,
        "Track": 3,
        "Name": "EngCustCC"
      }
    ]
```

**Comment by quebecman** [comment]
Most of the time, I drag it not in the QUEUE but directly in the main windows since I need to apply some minor modification such as SUBTITLE NAME, audio name (wich would be nice in a picker (batch naming).

After that, I click queue and once I have 5 to 10 files, I click encode.

Thanx. Regards.

________________________________

**Comment by RandomEngy** [comment]
I was asking @dan64, who was having the issue.

**Comment by quebecman** [comment]
Lol... my bad, didn't knew I was receiving emails from someone else.

**Comment by dan64** [comment]
I'm doing the same steps: drag & drop and name change. Then I press the Arrow button to encode or in the case there is already an encoding task active, I press the button to add the encoding job to the queue.

**Comment by RandomEngy** [comment]
Thanks for the sample source. I found the issue and it will be fixed in the next Beta release.

**Comment by RandomEngy** [comment]
Fixed in 6.23 Beta.
